### DISCUSSION TITLE
Scripting Documentation

### DISCUSSION THREAD

**Discussion by sifran-github** [freeplane/docs#21] [General]
Sorry for the delay. I just sent a pull request with scripting documentation start.
I have never used github or docsify. i'm not completely sure how to work with it.
Do i need to fork -> edit it locally -> pull request it back? every time i edit something? if so, how can i keep the latest version?
Also - how do i create different pages that link to each other? do i need to install docsify locally? or can i just create multiple md files on GitHub website?

Anyway, I just made a pull request for the scripting page, containing some simple examples.

Regarding the API Documentation (in Freeplane: Help->Freeplane API): is it complete? is it automatically generated? or is it just docs that someone wrote? how do you think that we should combine it with our scripting documentation?

At the moment i just put everything in one file... but of course we need to expand it to different sections.
I feel that i need help and better understanding of our documentation structure in order to contribute.

**Comment by sifran-github** [comment]
I suggest to have a meeting just about Documentation, so all of the questions that i asked can be answered and explained to people who want to contribute (including myself). the meeting should include at least:
1. Go over the flow of "how to contribute and post changes to the documentation". including someone's screen doing it.
2. Explaining the roadmap of converting the old wiki to a new one & Explaining to people how they can contribute by taking parts of the docs into the new wiki. (do we know how to do it and mark which pages/sections have been moved?)

The last meeting was more about getting ideas from everyone, but i think we should have a meeting where someone should teach the participants how it works and how to contribute. More of a meeting to teach others.

**Reply by macmarrum** [reply]
> I just sent a pull request with scripting documentation start.

That's great! 😄 

> I have never used github or docsify. i'm not completely sure how to work with it.

I understand that working with GitHub can be intimidating at the beginning. I can see your Pull Request, so whatever you did has had a good outcome 👍 

> Do i need to fork -> edit it locally -> pull request it back? every time i edit something? if so, how can i keep the latest version?

The fork action is required only once, at the very beginning → https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/about-forks
From now on, you will only sync your fork (copy) of the upstream freeplane/wiki repository → https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork
And when you have new content, you will create a Pull Request to incorporate your changes to the upstream freeplane/wiki repo.

> Also - how do i create different pages that link to each other? do i need to install docsify locally? or can i just create multiple md files on GitHub website?

No need to install docsify. You simply need to create a new Markdown file (e.g. node.md) and use the standard Markdown syntax to link to it from index.md. You can do it in the GitHub web interface – button `Add file` → https://docs.github.com/en/repositories/working-with-files/managing-files/adding-a-file-to-a-repository

Please see also
* https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax
* https://github.github.com/gfm/

You can verify how the wiki will display by enabling GitHub Pages on your fork. Go to `Settings->Pages` (https://github.com/sifran-github/wiki/settings/pages) and enable it for `main` branch, `/ (root)` folder. You will then be able to see how docsify displays your content on https://sifran-github.github.io/wiki/

**Reply by sifran-github** [reply]
thank you! i will check all the links.

Somehow my GitHub Pages worked only when I defined it to use the /docs directory and not the / (root), although I don't have a /docs directory... im not sure why but it works...
